opm/simulators/timestepping/TimeStepControl.cpp

```diff
@@ -141,17 +142,23 @@ namespace Opm
         errors_[ 2 ] = error;
         for( int i=0; i<2; ++i ) {
             assert(std::isfinite(errors_[i]));
-            assert(errors_[i]>0);
+            assert(errors_[i]>=0);
         }
 
-        if( error > tol_ )
+        if( errors_[2] > tol_ )
         {
             // adjust dt by given tolerance
             const double newDt = dt * tol_ / error;
             if( verbose_ )
                 std::cout << "Computed step size (tol): " << unit::convert::to( newDt, unit::day ) << " (days)" << std::endl;
             return newDt;
         }
+        else if (errors_[1] * errors_[2] == 0.)
```

Comment:
Exact floating point equality comparisons, even with zero, always make me worry that we're missing something.  Is there no other way of phrasing this condition?

Reply:
I know, it looks worrying. But technically, it really only causes problem when one of these two values are exactly 0.  If it is some extremely small values, the results will be the same with the current approach (most likely, while not excluding extraordinarily designed  situations).  

Alternatively, maybe we can use something like the following as the if condition here:
```c++ 
if (errors_[1] < std::numeric_limits<double>::min() || 
     errors_[2] < std::numeric_limits<double>::min()) 
``` 
Like me know what do you think.